### PR TITLE
Lint: Disable errors for formatting handled by prettier

### PR DESCRIPTION
**Pull request by blowery**
Since we now use prettier to format all of our code, we can turn eslint formatting errors covered by prettier (max-len, indent...). Prettier provides a config to do just that in `eslint-format-prettier`.

~The only real problem we have now is ternary expressions, so we could just[ ignore those nodes instead](https://eslint.org/docs/rules/indent#ignorednodes).~

See #24504, #24571

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,40 +1,69 @@
...
-		'jsx-a11y',
-		'import',
-	],
+	plugins: [ 'jest', 'jsx-a11y', 'import' ],
 	rules: {
-		camelcase: 0, // REST API objects include underscores
+		// REST API objects include underscores
+		camelcase: 0,
+
+		// TODO: why did we turn this off?
```
Comment (gziolo) [inline .eslintrc.js]: I think it errors because there are still `chai` matchers which aren't compatible with Jest syntax.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,40 +1,69 @@
...
-		'plugin:jsx-a11y/recommended',
-		'plugin:jest/recommended',
-	],
+	extends: [ 'wpcalypso/react', 'plugin:jsx-a11y/recommended', 'plugin:jest/recommended' ],
 	parser: 'babel-eslint',
 	env: {
 		browser: true,
 		'jest/globals': true,
+		// mocha is only still on because we have not finished porting all of our tests to jest's syntax
 		mocha: true,
```
Comment (gziolo) [inline .eslintrc.js]: You can try to turn it off as we most likely migrated all globals to Jest.

Comment (blowery) [reply .eslintrc.js]: sure, different PR for that though.

**Comment by blowery** [comment]
ok, now we really really use prettier for everything

https://github.com/Automattic/wp-calypso/pull/24752

thanks @sirreal!

File: .eslintrc.js [hunk 3]
```diff
@@ -1,40 +1,69 @@
...
+		// TODO: why did we turn this off?
 		'jest/valid-expect': 0,
-		'jsx-a11y/anchor-has-content': 0, // i18n-calypso translate triggers false failures
+
+		// i18n-calypso translate triggers false failures
+		'jsx-a11y/anchor-has-content': 0,
+
+		// error if any given line is over 140 characters long.
+		// Ideally this would be even lower(120, 80?), but that's a big step.
 		'max-len': [ 2, { code: 140 } ],
```
Comment (sirreal) [inline .eslintrc.js]: Shall we disable this as well? Prettier's functionality is based on breaking long lines.

[A quick search](https://github.com/Automattic/wp-calypso/search?utf8=%E2%9C%93&q=max-len&type=) will show lots of disabling for SVG paths or strings that aren't broken.

File: .eslintrc.js [hunk 4]
```diff
@@ -1,40 +1,69 @@
...
+		// error if any module depends on the data-observe mixin, which is deprecated
 		'no-restricted-imports': [ 2, 'lib/mixins/data-observe' ],
 		'no-restricted-modules': [ 2, 'lib/mixins/data-observe' ],
-		'no-unused-expressions': 0, // Allows Chai `expect` expressions
-		'wpcalypso/jsx-classname-namespace': [ 2, {
-			rootFiles: [ 'index.js', 'index.jsx', 'main.js', 'main.jsx' ],
-		} ],
+
+		// Allows Chai `expect` expressions. Now that we're on jest, hopefully we can remove this one.
+		'no-unused-expressions': 0,
```
Comment (sirreal) [inline .eslintrc.js]: No change, layout and comment.

File: .eslintrc.js [hunk 5]
```diff
@@ -1,40 +1,69 @@
...
+
+		// enforce our classname namespacing rules
+		'wpcalypso/jsx-classname-namespace': [
+			2,
+			{
+				rootFiles: [ 'index.js', 'index.jsx', 'main.js', 'main.jsx' ],
+			},
+		],
```
Comment (sirreal) [inline .eslintrc.js]: No change, layout and comment.

File: .eslintrc.js [hunk 6]
```diff
@@ -1,40 +1,69 @@
...
+		// - path because we use it quite a bit
 		'import/no-nodejs-modules': [ 'error', { allow: [ 'url', 'events', 'path' ] } ],
+
+		// Disallow importing or requiring packages that are not listed in package.json
+		// This prevents us from depending on transitive dependencies, which could break in unexpected ways.
 		'import/no-extraneous-dependencies': [ 'error', { packageDir: './' } ],
-	}
+
+		// turn off indenting rules since we let prettier handle it
+		indent: 0,
```
Comment (sirreal) [inline .eslintrc.js]: New rule disabling eslint indent 🎉

File: .eslintrc.js [hunk 7]
```diff
@@ -1,40 +1,69 @@
...
-		camelcase: 0, // REST API objects include underscores
+		// REST API objects include underscores
+		camelcase: 0,
+
+		// TODO: why did we turn this off?
 		'jest/valid-expect': 0,
-		'jsx-a11y/anchor-has-content': 0, // i18n-calypso translate triggers false failures
+
+		// i18n-calypso translate triggers false failures
+		'jsx-a11y/anchor-has-content': 0,
```
Comment (sirreal) [inline .eslintrc.js]: No change, layout.

**Review by sirreal — COMMENTED** [review]
Excited to remove an annoyance with this 🙂 

Added a comment about `max-len`. If our goal here is "`prettier` handles that, be silent `eslint`" PR, then I'd like to remove `max-len` as well.

Commented a few of the confusing parts of the diff so it's clear what rules aren't changed but have been formatted.

**Comment by sirreal** [comment]
I've pushed a commit that removes the now-redundant `indent` disable/enable comments. Feels so good!

File: .eslintrc.js (hunk 3 above)
Comment (blowery) [reply .eslintrc.js]: sounds good to me. I'll rename this PR.

**Comment by blowery** [comment]
@sirreal taking a slightly different approach here. I'm going to pull in and use https://github.com/prettier/eslint-config-prettier

**Comment by blowery** [comment]
```
--- before	2018-05-11 10:28:38.000000000 -0400
+++ after	2018-05-11 10:28:50.000000000 -0400
@@ -1,11 +1,9 @@
-indent: 1790
 wpcalypso/jsx-classname-namespace: 844
 prefer-const: 647
 react/no-string-refs: 249
 valid-jsdoc: 247
 jsx-a11y/click-events-have-key-events: 106
 jsx-a11y/no-static-element-interactions: 98
-max-len: 98
 jsx-a11y/anchor-is-valid: 69
 jsx-a11y/alt-text: 62
 jsx-a11y/label-has-for: 41
@@ -38,14 +36,12 @@
 react/no-did-mount-set-state: 3
 react/no-is-mounted: 3
 jsx-a11y/media-has-caption: 2
-no-multi-spaces: 2
 no-var: 2
 eqeqeq: 1
 jsx-a11y/html-has-lang: 1
 jsx-a11y/role-has-required-aria-props: 1
 no-nested-ternary: 1
 no-use-before-define: 1
-operator-linebreak: 1
 wpcalypso/i18n-no-variables: 1

-total: 4585
+total: 2694
```

**Comment by samouri** [comment]
> Disable errors for formatting handled by prettier

This PR is doing a lot more than that right?

**Review by samouri — APPROVED** [review]
Two small changes, but after fixing those feel free to 🚢

File: client/lib/shortcode/index.js [hunk 8]
```diff
@@ -7,8 +7,8 @@ import { isEqual, memoize } from 'lodash';
 /**
  * Module variables
  */
-const REGEXP_ATTR_STRING = /(\w+)\s*=\s*"([^"]*)"(?:\s|$)|(\w+)\s*=\s*\'([^\']*)\'(?:\s|$)|(\w+)\s*=\s*([^\s\'"]+)(?:\s|$)|"([^"]*)"(?:\s|$)|(\S+)(?:\s|$)/g; // eslint-disable-line max-len
-const REGEXP_SHORTCODE = /\[(\[?)([^\[\]\/\s\u00a0\u200b]+)(?![\w-])([^\]\/]*(?:\/(?!\])[^\]\/]*)*?)(?:(\/)\]|\](?:([^\[]*(?:\[(?!\/\2\])[^\[]*)*)(\[\/\2\]))?)(\]?)/; // eslint-disable-line max-len
+const REGEXP_ATTR_STRING = /(\w+)\s*=\s*"([^"]*)"(?:\s|$)|(\w+)\s*=\s*\'([^\']*)\'(?:\s|$)|(\w+)\s*=\s*([^\s\'"]+)(?:\s|$)|"([^"]*)"(?:\s|$)|(\S+)(?:\s|$)/g;
+const REGEXP_SHORTCODE = /\[(\[?)([^\[\]\/\s\u00a0\u200b]+)(?![\w-])([^\]\/]*(?:\/(?!\])[^\]\/]*)*?)(?:(\/)\]|\](?:([^\[]*(?:\[(?!\/\2\])[^\[]*)*)(\[\/\2\]))?)(\]?)/;
```
Comment (sirreal) [inline client/lib/shortcode/index.js]: 😵

**Review by sirreal — APPROVED** [review]
I cleaned up the empty braces @samouri spotted.

Looked through code, seems correct.
Smoke tested, didn't spot any errors.

Let's 🚢 !

**Pull request merged**